core/src/main/scala/org/ensime/core/Project.scala

```diff
@@ -12,10 +12,13 @@ import org.ensime.indexer._
 
 import scala.collection.immutable.ListSet
 import scala.concurrent.duration._
+import scala.util.Properties._
 import scala.util._
 import org.ensime.util.file._
 import org.ensime.util.FileUtils
 
+case object CloseOnIndexUpdate
```

Comment:
I cannot directly send `ShutdownRequest` as it is defined in server module and not visible in core. Without the case object what option do I have to send callback